ODIN_II/SRC/ast_util.cpp

```diff
@@ -399,6 +399,31 @@ int get_range(ast_node_t* first_node)
 	return -1; // indicates no range
 }
 
+/*---------------------------------------------------------------------------------------------
+ * (function: get_range_part_select)
+ * 	Check the node range is legal. Will return the range if it's legal.
+ *  Node should have three children. Second and Third children's type should be NUMBERS.
+ *  Direction parameter differentiates between ascending/descending part selector (1 for
+ *  ascending and -1 for descending selection).
+ *-------------------------------------------------------------------------------------------*/
+int get_range_part_select(ast_node_t *first_node, char direction)
```

Comment:
could you use the error_message(...) function here to display a verbose error to explain to the user that we only support [High number: Low number] ranges. oassert should be avoided when possible so that the user gets some sense of why things do not compile. oassert are used to assert that a function does receive the right parameter from another function and allows the developper to debug the error. oassert are NOT a way to tell a user that some form of verilog is not supported.